soupy_status.py

```diff
@@ -0,0 +1,115 @@
+"""
+All HTTP Status Codes as per:
+https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
+"""
+
+__license__ = 'GPL-3.0+'
+
+from enum import IntEnum
+
+class SoupyStatus(IntEnum):
```

Comment:
I'm confused why you made this list when `Soup.Status` already exists?

Reply:
> I'm not sure it makes sense to maintain our own list.

Ideally we wouldn't have to maintain a list of our own. But it's not a huge list and I'll search for or file bugs to see if their list could be expanded. It kinda baffled me as to why it's missing some official/widely used status codes. But in the mean time this is a better solution otherwise you'll not get real status codes from Pandora.

> maybe even extend the list upstream in Soup?

I'm not sure if or how you would "extend" an Enum. It's that kinda the point that they are constants and can't be changed?

Reply:
> I'm not sure if or how you would "extend" an Enum. That's kinda the point that they are constants and can't be changed?

Nobody was suggesting changing values, obviously the HTTP spec didn't change values.

Just duplicate what you did here in libsoup: https://github.com/GNOME/libsoup/blob/master/libsoup/soup-status.h#L38-L93

Reply:
> Just duplicate what you did here in libsoup

That's what I did...

Reply:
Do you want verbose descriptions of the status codes? I can do that if you like?

Reply:
> That's what I did...
 
No you moved logic that belongs in libsoup into here. Which I understand is the short term solution.

> Do you want verbose descriptions of the status codes? I can do that if you like?

No, that wasn't my point.

Reply:
I guess I don't understand what you want then. soupy_status.py contains all of the official status codes plus all of the widely used non-spec status codes.

Reply:
`soupy` exists for one reason; To make libsoups C API more pythonic. All I am saying is adding features such as an enum of status codes doesn't make sense here, that makes sense in libsoup.

Reply:
OK, I assume you mean I should submit a patch to libsoup?

Reply:
Yes.

Reply:
Yep, I've never submitted a patch let alone wrote any C, although in this case it looks pretty trivial(a little copy pasta). The patch stuff would prob be the hardest part for me to figure out and I'm sure it's not that hard. Some advice and maybe a short how-to for the patch would help if you would?

Reply:
Well step one is just getting it to build, `dnf builddep libsoup`, `./autogen.sh`, `make -sj6`. Since its only modifying an enum and not any logic if it builds it is probably correct.

A patch file can be generated from your commit with `git format-patch HEAD~1` and thrown on https://bugzilla.gnome.org

Feel free to have me review it before or after you open the bug.

Reply:
Thanks I will.

Reply:
Hey @TingPing I was looking at the emums and it appears that in a few cases they have multiple statuses assigned to the same status code number. That doesn't look right to me?

Reply:
I was thinking for the patch I would use https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml as the source. Seems pretty authoritative to me. And I see no reason why multiple statuses should share the same status number. Would that even work? I'm not exactly sure enumes work in C, but it seems to me that it wouldn't work as expected. In my mind it should be one status per status number?

Reply:
Sounds like a mistake. An enum in C doesn't enforce unique-ness though.

Reply:
Anyway it can't be changed now its forever part of its API.

Reply:
This is what I'm talking about:
https://github.com/GNOME/libsoup/blob/master/libsoup/soup-status.h#L70

Again I'm not sure but I would read that to mean that 3 statuses share the same status code?

And this:
https://github.com/GNOME/libsoup/blob/master/libsoup/soup-status.h#L54-L55

2 statuses that share the same status code?

If this in fact can't be changed and it is a mistake that means the enum is forever broken and there's no point using it...

Reply:
It certainly defeats the purpose of an enum...

```python
import gi
gi.require_version('Soup', '2.4')
from gi.repository import Soup

print(Soup.Status.FOUND is Soup.Status.MOVED_TEMPORARILY)
```

```>>>True```